### PR TITLE
feat(msteams): Assigned and unassigned activity notifications

### PR DESCRIPTION
**Pull request by 97varun**
Add assigned and unassigned notifications.
- A new message builder to add the issue details.
- Tests for both types.

### CODE REVIEW

**Review by NisanthanNanthakumar — APPROVED** [review]

File: src/sentry/integrations/msteams/notifications.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                for channel, integration in integrations_by_channel.items():
+                    conversation_id = get_user_conversation_id(integration, channel)
+
+                    client = MsTeamsClient(integration)
+                    try:
+                        client.send_card(conversation_id, card)
+                    except Exception as e:
+                        logger.error(f"Exception occured while trying to send the notification {e}")
+
+    metrics.incr(
```
Comment (scefali) [inline src/sentry/integrations/msteams/notifications.py:96]: @97varun shouldn't we also call `record_notification_sent`?

File: src/sentry/notifications/defaults.py [hunk 2]
```diff
@@ -34,7 +34,22 @@
     NotificationSettingTypes.QUOTA_WARNINGS: NotificationSettingOptionValues.ALWAYS,
 }
 
+NOTIFICATION_SETTINGS_MSTEAMS = {
+    NotificationSettingTypes.DEPLOY: NotificationSettingOptionValues.NEVER,
+    NotificationSettingTypes.ISSUE_ALERTS: NotificationSettingOptionValues.NEVER,
+    NotificationSettingTypes.ACTIVE_RELEASE: NotificationSettingOptionValues.NEVER,
+    NotificationSettingTypes.WORKFLOW: NotificationSettingOptionValues.NEVER,
+    NotificationSettingTypes.APPROVAL: NotificationSettingOptionValues.NEVER,
```
Comment (scefali) [inline src/sentry/notifications/defaults.py:42]: @97varun Certain settings should just be enabled by default such as `NotificationSettingTypes.APPROVAL`, you can follow what we do for Slack.

**Review by scefali — CHANGES_REQUESTED** [review]

**Review by scefali — COMMENTED** [review]
@97varun How are we gating this feature to make sure folks don't get MS teams notifications until this feature is ready to launch?

**Comment by 97varun** [comment]
> @97varun How are we gating this feature to make sure folks don't get MS teams notifications until this feature is ready to launch?

@scefali As long as the UI changes (https://github.com/getsentry/sentry/pull/36167) aren't made available to the user they can't enable the notification settings for MS Teams and this feature remains inaccessible.

As far as the personal installation flow goes (https://github.com/getsentry/sentry/pull/36749). We couldn't think of a way to hide this behind a feature flag as any MS Teams user (even without a sentry account) could trigger it. Enabling that would be like flipping a switch for everyone.

From what we discussed so far, we were planning to implement a few notifications and then make it available to everyone. Do you have any suggestions on how to gate this feature given these constraints?

File: src/sentry/notifications/defaults.py (hunk 2 above)
Comment (97varun) [reply src/sentry/notifications/defaults.py]: @scefali These notifications aren't implemented and enabling these types before deploying the UI changes makes the settings UI behave weirdly, so I have kept them disabled till we introduce the UI changes and support these notification types.

**Review by scefali — APPROVED** [review]

**Comment by scefali** [comment]
> @scefali As long as the UI changes (https://github.com/getsentry/sentry/pull/36167) aren't made available to the user they can't enable the notification settings for MS Teams and this feature remains inaccessible.

@97varun We want some notifications enabled by default. However, I don't think you are actually going to implement any of those ones that are on by default. But it should be relatively trivial to check a feature flag before sending any ms teams notifications after installation.

**Comment by 97varun** [comment]
> > @scefali As long as the UI changes (#36167) aren't made available to the user they can't enable the notification settings for MS Teams and this feature remains inaccessible.
> 
> @97varun We want some notifications enabled by default. However, I don't think you are actually going to implement any of those ones that are on by default. But it should be relatively trivial to check a feature flag before sending any ms teams notifications after installation.

@scefali Yes, I could send the notifications if a feature flag is set but I want to send notifications to users who see this message on installation and expect to receive notifications. And, I don't see a way to feature flag the installation change.

image

**Pull request merged**